programs/gzip.c

```diff
@@ -576,6 +591,11 @@ tmain(int argc, tchar *argv[])
 			 *  option as a no-op.
 			 */
 			break;
+		case 't': /// test
+			options.test = true;
+			options.decompress = true;
+			options.to_stdout = true;
```

Comment:
It's not obvious why `to_stdout` is being set here.  After all, `-t` doesn't actually write anything to standard output.  It would be helpful to add a comment that properly explains this:

                        /*                                                       
                         * -t behaves just like the more commonly used -c        
                         * option, except that -t doesn't actually write         
                         * anything.  For ease of implementation, just pretend   
                         * that -c was specified too.                            
                         */                                                      
                        options.to_stdout = true;                                

Either that or properly abstract this out into something like `options.no_output_file`.  I'm leaning towards just reusing `to_stdout` as you've done, but it needs to be properly explained.